### ISSUE TITLE
[REG 2.101-master] C header files have precedent over D modules in imports

### ISSUE THREAD

**Issue by dlangBugzillaToGithub** [dlang/dmd#18142]
### CC List

* Walter Bright (@WalterBright)
* Iain Buclaw (@ibuclaw)

### Description

````markdown
This test fails.

$ dmd -I=imports importd.d
importd.d(2): Error: undefined identifier `Have_Qsort_R`
importd.d(2):        while evaluating: `static assert(Have_Qsort_R)`


importd.d
```
import gcc.config;
static assert(Have_Qsort_R);
```

imports/gcc/config.d
```
module gcc.config;
enum Have_Qsort_R = true;
```

gcc/config.h
```
#ifndef GCC_CONFIG_H
#define GCC_CONFIG_H
#ifdef GENERATOR_FILE
#error config.h is for the host, not build, machine.
#endif
#ifdef IN_GCC
# include "ansidecl.h"
#endif
#endif /* GCC_CONFIG_H */
```
````

**Comment by dlangBugzillaToGithub** [comment]
#### ibuclaw (@ibuclaw) commented on 2022-12-10T14:49:51Z



````markdown
Caused by fix for issue 23479
````

**Comment by dlangBugzillaToGithub** [comment]
#### ibuclaw (@ibuclaw) commented on 2022-12-10T14:50:15Z



````markdown
https://github.com/dlang/dmd/pull/14636
````

**Comment by dlangBugzillaToGithub** [comment]
#### ibuclaw (@ibuclaw) commented on 2022-12-30T23:38:56Z



````markdown
*** Issue 23548 has been marked as a duplicate of this issue. ***
````

**Comment by dlangBugzillaToGithub** [comment]
#### bugzilla (@WalterBright) commented on 2023-01-10T06:07:36Z



````markdown
The config.h file is found first because config.h appears before config.d in the import search path.

If config.h and config.d were in the same directory, the .d file would take precedence.
````

**Comment by Geod24** [comment]
@thewilsonator : Which PR fixed this ? #14636 introduced the regression but I'm not sure it's been fixed ?

**Comment by Geod24** [comment]
https://github.com/dlang/dmd/pull/14870 I guess ?
